# Homework 3

## Your Name Here (or your names here if you are pair programming)

 - Student Name: Andre Sae
 - Student UT EID: as226576


 - Partner Name:
 - Partner UT EID:


## Normal and t distributions
## t-test


An experiment was conducted to determine the effect of children participating in a given meal preparation on calorie intake for that meal. Data are recorded below. 

Save the data to a format that can be read into python. Read the data in for analysis. Data is provided in two separted CSV files. 

* Use python to calculate the quantities and generate the visual summaries requested below. You will lose points if you are not utilizing python.

* You can use scipy libary or other libraries to do your tests or you can implement them from scratch in python 



In [2]:
# Standard Headers
# You are welcome to add additional headers here if you wish
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# Enable inline mode for matplotlib so that Jupyter displays graphs
%matplotlib inline

# Question - 1
Summarize the data by whether children participated in the meal preparation or not. Use an appropriately labelled table to show the results. Also include a graphical presentation that shows the distribution of calories for participants vs. non-participants. Describe the shape of each distribution and comment on the similarity (or lack thereof) between the distributions in each population. **(2 points)**

Be aware that there is not one specific way the graph needs to look. Experiment with different types of graphs, and different parameters for the graph type. Your goal is to present the data as readable as possible. 

In [2]:
# Add your code or descriptions here 

# Question - 2 

Does the mean calorie consumption for those who participated in the meal preparation differ from **425**? Formally test at the $\alpha = 0.05$ level using the 5 steps outlined in the module. **(6 points)**


In [3]:
# Add your code or descriptions here 

# Question -3 
Calculate a **90%** confidence interval for the mean calorie intake for participants in the meal preparation. Interpret the confidence interval. **(6 points)**

In [6]:
from scipy import stats
import math

particpants = pd.read_csv('participants.csv')


In [10]:
mean, sigma = np.mean(particpants), np.std(particpants)

stats.norm.interval(0.90, loc=mean, scale=(sigma / math.sqrt(particpants.size)))

/home/andre/.local/src/cs329e_analysis/.venv/lib/python3.13/site-packages/numpy/_core/fromnumeric.py:4062: FutureWarning: The behavior of DataFrame.std with axis=None is deprecated, in a future version this will reduce over both axes and return a scalar. To retain the old behavior, pass axis=0 (or do not pass axis)
  return std(axis=axis, dtype=dtype, out=out, ddof=ddof, **kwargs)


(array([370.91277009]), array([449.24642991]))

Given 25 sample draw, we are 90% confident that the mean is within 370.912 and 449.246.

# Question 4 
Formally test whether or not participants consumed 
more calories than non-participants at the $\alpha = 0.05$ level using the 5 steps procedure for hypothesis tests. **(6 points )**

Our hypothesis:
H_0: The participant mean calories is larger than the non partcipant mean calories
H_\alpha: The particpant mean calories is the same or smaller than the non participant mean calories

Our test:
We need to use a t-test because our sample size is smaller than 30 and our population standard deviation is unknown.

In [13]:
from scipy import stats 

t_stat, p_val = stats.ttest_ind(
    a = pd.read_csv('participants.csv'),
    b = pd.read_csv('nonparticipants.csv'),
    alternative='less',
    equal_var=False
)

alpha = 0.05

print("p-val: " + str(p_val))

if p_val < alpha: 
    print("The null hypothesis is rejected because the p-val from the two sample t-test is smaller than tha alpha.")

elif p_val >= alpha:
    print("The null hypothesis was not rejected because the p-val from the two sample t-test was the same or larger than the two sample t-test alpha value.")

p-val: [0.82967751]
The null hypothesis was not rejected because the p-val from the two sample t-test was the same or larger than the two sample t-test alpha value.
